src/indexer.c

```diff
@@ -290,21 +290,40 @@ static void writeMissingFieldDocs(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx)
     if (!found_df) {
       InvertedIndex *iiMissingDocs = dictFetchValue(spec->missingFieldDict, fs->name);
       if(iiMissingDocs == NULL) {
-        size_t dummy_mem;
-        iiMissingDocs = NewInvertedIndex(Index_DocIdsOnly, 1, &dummy_mem);
+        size_t index_size;
+        iiMissingDocs = NewInvertedIndex(Index_DocIdsOnly, 1, &index_size);
+        aCtx->spec->stats.invertedSize += index_size;
         dictAdd(spec->missingFieldDict, fs->name, iiMissingDocs);
       }
       // Add docId to inverted index
       t_docId docId = aCtx->doc->docId;
       IndexEncoder enc = InvertedIndex_GetEncoder(Index_DocIdsOnly);
       RSIndexResult rec = {.type = RSResultType_Virtual, .docId = docId, .offsetsSz = 0, .freq = 0};
-      InvertedIndex_WriteEntryGeneric(iiMissingDocs, enc, docId, &rec);
+      aCtx->spec->stats.invertedSize += InvertedIndex_WriteEntryGeneric(iiMissingDocs, enc, docId, &rec);
     }
   }
 
   dictRelease(df_fields_dict);
 }
 
+// Index the doc in the existing docs inverted index
+static void writeExistingDocs(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx) {
+  if (sctx->spec->rule && !sctx->spec->rule->index_all) {
```

Comment:
Why we need to check for `sctx->spec->rule` is not NULL?

Reply:
I think the unit-tests may invoke this function without a rule, will check it out

Reply:
@MeirShpilraien Yes, indeed we reach here only from our unit-tests.
Specifically: `testAddDocumentTextField` in `test_cpp_llapi.cpp:47` creates an index via the `RediSearch_CreateIndex` API with `options = NULL` such that it is substituted with a default that does not set the score or lang fields. Then, we create a `spec->rule` only if `options->score` or `options->lang` was set - such that in this case we don't have a rule.